TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -150,17 +43,71 @@ class BenchmarkParser extends Parser {
+            isValid = false;
+        }
 
+        if ( isValid ) {
+            curBenchVariant = BenchmarkMetric[BenchmarkMetricRouter[benchmarkName][benchmarkVariant]];	
+                
+            // if outerRegex is undefined, all runs should be measured. Parse metrics in every run
+            // if outerRegex is defined, any runs before outerRegex will be ignored. Parse metrics in warm runs only 
+            if (curBenchVariant.outerRegex !== undefined) {
+                if ( ( curRegexResult = curBenchVariant.outerRegex.exec( testOutput ) ) !== null ) {
+                    // index 0 contains entire text (curItr.value)
+                    // index 1 contains text after the outerRegex
+                    testOutput = curRegexResult[1];
+                } 
+            }
+                
+            // Parse metric values
+            curTestData["metrics"] = [];
+            let curMetricList = Object.keys(curBenchVariant["metrics"]);
+            for ( let i = 0; i < curMetricList.length; i++ ) {
+
+                curMetric = curMetricList[i];
+                curRegex = curBenchVariant["metrics"][curMetric]["regex"];  
+                /*	Parse all values for single metric from result to an array
+                 *  e.g 
+                 *  Liberty Startup =>
+                 *  curRegexResult = ['startup time in ms 32',32,'startup time in ms 41',41, x 6] 
+                 *  Liberty Throughput =>
+                 *  curRegexResult = ['<metric type="throughput">32<\/data>',32]
+                 */
+                curRegexResult = testOutput.split(curRegex);
+                //collect only the capture groups from the regex array
+                curRegexResult = curRegexResult.filter( (value,index) => (index %2 === 1));
+                curMetricValues = curRegexResult.map(parseFloat);
+                /* Metrics such as JITCPU total in LibertyThroughput and geomean_GCA in CryptoBB, 
+                 * require aggregate function to be applied. 
+                 * e.g JIT CPU total values are collected as [ 0,0,300,200,20 ] but we display 520 as the JITCPU total  
+                 */
+                if(typeof curBenchVariant["metrics"][curMetric]["funcName"] != "undefined" && curMetricValues.length != 0) {
+                    curMetricValues = [curBenchVariant["metrics"][curMetric]["funcName"](curMetricValues)];
+                }
+                curTestData["metrics"].push({name: curMetric, value: curMetricValues});
+            }     
+        }
         return {
-            tests,
-            jdkDate,
-            javaVersion,
-            buildResult,
-            machine: this.extractMachineInfo( output ),
-            type: "Perf",
-            startBy: this.extractStartedBy( output ),
-            artifactory: this.extractArtifact( output ),
+            benchmarkName,
+            benchmarkVariant,
+            perfResult: isValid ? "PASSED" : "FAILED",
```

Comment:
Don't think we need this. Maybe we can just use `testResult`. Otherwise, it would be confusing to have both of them.